### DISCUSSION TITLE
Extract Timestamp with minutes part only

### DISCUSSION THREAD

**Discussion by ShivaniSK** [finos/TimeBase-CE#230] [Q&A] [answered]
Hi,
Currenty we have stored timestamp in Timebase as below

```
0,2022-08-26 20:55:12.123
1,2022-08-26 20:56:23.222
2,2022-08-26 20:57:11.111
3,2022-08-26 20:58:12.123
4,2022-08-26 20:59:13.666
5,2022-08-26 21:00:55.111
6,2022-08-26 21:01:12.665
```
Is there any timebase query which can give the output as below?

```
0,2022-08-26 20:55:00
1,2022-08-26 20:56:00
2,2022-08-26 20:57:00
3,2022-08-26 20:58:00
4,2022-08-26 20:59:00
5,2022-08-26 21:00:00
6,2022-08-26 21:01:00
```
 
Is there any function which can give me output as above with timestamp having minutes data?

**Comment by alex-karpovich** [comment]
Greetings, @ShivaniSK 
you can use additional column only

select timestamp - (timestamp AS INT64 % 60000) as time from "BINANCE"

**Reply by ShivaniSK** [reply]
Hi @alex-karpovich  the above query is giving below error 
`Error: CompilationException: 1.21..34: CompilationException: Can't cast TIMESTAMP to INTEGER`

**Comment by alex-karpovich** [comment]
with (timestamp AS INT64) as time
select (time - (time % 60000) AS timestamp) as 'time' from "BINANCE"

**Reply by ShivaniSK** [reply]
Hi @alex-karpovich , on executing above query in shell its giving below error:
`>>> Error: WITH: unrecognized command. (Type ? for usage)`

**Comment by alex-karpovich** [comment]
to be executed in TimeBase shell use ? before statements

> ? with (timestamp AS INT64) as time select (time - (time % 60000) AS timestamp) as 'time' from "BINANCE" 

from help
```
*******************************************************************************
*                      QuantQuery Language                                    *
*******************************************************************************

? <qql>                     Run a QQL query or command
```

**Reply by ShivaniSK** [reply]
Hello @alex-karpovich , using "?" before the WITH statements in shell resolved the "WITH command unrecognized error". However its giving below error now when executing the below query
`? with (timestamp AS INT64) as time select (time - (time % 60000) AS timestamp) as 'time' from "BINANCE"`
`>>> Error: CompilationException: 1.7..20: CompilationException: Can't cast TIMESTAMP to INTEGER
`

**Reply by alex-karpovich** [reply]
ok. let me publish updated version

**Reply by ShivaniSK** [reply]
sure thanks @alex-karpovich . Can you please let us know the version once its published? We can then plan our development timelines according to that.

**Reply by ShivaniSK** [reply]
Hi @alex-karpovich , do we have this fix published in 6.1.3 version?

**Reply by alex-karpovich** [reply]
It will be available in 6.1.14. Optimistic ETA is Monday, 5 May

**Reply by ShivaniSK** [reply]
Hello @alex-karpovich , the new version available now?

**Reply by ShivaniSK** [reply]
Hi @alex-karpovich , do we have any update on the new version?
Incase the date for publishing the version is finalized, please let us know so that we can plan our deliveries accordingly.
Appreciate your support :)

Regards,
Shivani

**Reply by alex-karpovich** [reply]
Greetings @ShivaniSK. was totally busy with other things. apologies. Next Monday I release new build

**Reply by ShivaniSK** [reply]
Hello @alex-karpovich the new build is expected to be released this week or next?

**Reply by alex-karpovich** [reply]
Greetings, on this week.

**Reply by ShivaniSK** [reply]
Hello @alex-karpovich do we have the new build available now?

**Reply by alex-karpovich** [reply]
I'll publish release tomorrow. Changes in repository, need verififcation

**Reply by alex-karpovich** [reply, marked as answer]
published https://github.com/finos/TimeBase-CE/releases/tag/6.1.14

**Reply by ShivaniSK** [reply]
thanks alot :)
